internal/env.go

```diff
@@ -32,3 +38,15 @@ func IsDatadogConfigured() bool {
 	}
 	return false
 }
+
+// IsDatadogEnabled checks if the Datadog integration is enabled. The
+// environment variable DD_ENABLE is checked. If the variable is missing the
+// Datadog integration is assumed to be enabled.
+func IsDatadogEnabled() bool {
```

Comment:
Logically strange, func `IsDatadogEnabled` returns `true` when the env variable DatadogEnable is not set or cannot be parsed as a boolean, which may unintentionally enable Datadog. A more cautious approach would be to return `false` in such cases, ensuring Datadog is only enabled with an explicit `true` value.

Reply:
As described in the body of the PR, this is done not break current implementation and ensure that Datadog does not have to be explicitly enabled. We can reverse this instead by naming the function `IsDatadogDisabled()`. This would probably fit better with the use-case.

@bendiknesbo what do you think.

Reply:
Renaming to `IsDatadogDisabled` could work.

@LinMAD, the introduction of the env-var to enabled/disable datadog will allow testing and development without Datadog set up locally.

Reply:
There is no question about naming, I refer to this algorithm, is that intended to be like?
```
	valStr := os.Getenv(DatadogEnable)
	val, err := strconv.ParseBool(valStr)
	if err != nil {
		return true
	}
	return val
```
Because `when the env variable DatadogEnable is not set or cannot be parsed as a boolean` as I wrote. Is that logically intended to be like that?

Reply:
I agree that it could confuse people, so I propose we change the implementation to explicitly disable Datadog instead. Unless the env variable is properly set Datadog should be enabled as it is currently is.

Reply:
I refactored this to focus on disabling Datadog, this way we do not change the production behaviour or expectation

Reply:
@LinMAD does this resolve your worries?

Reply:
Not exactly, since the code was updated and commented as a new thread for better consistency.